sylvia-derive/src/message.rs

```diff
@@ -609,11 +803,14 @@ impl<'a> GlueMessage<'a> {
                     let val = #sylvia ::serde_value::Value::deserialize(deserializer)?;
 
                     #(#deserialization_attempts)*
+                    #impl_deserialization_attempt
 
                     Err(D::Error::custom(format!(
-                        "Expected any of {} messages, but cannot deserialize to neither of those\n{}",
+                        "Expected any of {} or {} messages, but cannot deserialize to neither of those\n{}{}",
                         stringify!(#(#messages_type),*),
-                        [#(#deser_errors,)*].join("\n")
+                        stringify!(#impl_msg_name),
+                        [#(#deser_errors,)*].join("\n"),
+                        [#impl_deser_errors].join("\n")
```

Comment:
I am not a fan of the fact, that you are adding a new argument there. You can just add `impl_deser_errors` to the first array:

```suggestion
                        stringify!(#impl_msg_name),
                        [#(#deser_errors,)* #impl_deser_errors].join("\n")
```

This may be a bit cheaper, as you are joining on the single array (in particular - the second join is redundant as it is always one element array!). Obviously you also need to remove one `{}` from format string.